pkg/apis/core/types_shoot.go

```diff
@@ -303,6 +305,30 @@ type ClusterAutoscaler struct {
 	ScanInterval *metav1.Duration
 }
 
+// VerticalPodAutoscaler contains the configuration flags for the Kubernetes vertical pod autoscaler.
+type VerticalPodAutoscaler struct {
```

Comment:
Hm, no strong opinion. I'm fine with both, slightly prefer the current approach, mainly because it doesn't require the end-user to understand the architecture of VPA. He just needs to lookup the meaning of the flag, but he doesn't need to know which of the four components is responsible for executing it.
What do other folks think?

Reply:
Same. No strong opinions. Current approach does have an advantage that end-user need not bother which component is responsible for what, as mentioned by Rafael..
As long as the role of each parameter is documented properly, end-user need not care IMO.

Reply:
Then let's keep it as is. The names are enough and it is documented.